reactor-netty5-http/src/test/java/reactor/netty5/http/server/HttpServerTests.java

```diff
@@ -1913,6 +1917,327 @@ public void channelInboundEvent(ChannelHandlerContext ctx, Object evt) {
+		Http2SslContextSpec serverCtx = Http2SslContextSpec.forServer(ssc.certificate(), ssc.privateKey());
+		Http2SslContextSpec clientCtx =
+				Http2SslContextSpec.forClient()
+				                   .configure(builder -> builder.trustManager(InsecureTrustManagerFactory.INSTANCE));
+		doTestIssue1286(
+				server -> server.protocol(HttpProtocol.H2).secure(spec -> spec.sslContext(serverCtx)),
+				client -> client.protocol(HttpProtocol.H2).secure(spec -> spec.sslContext(clientCtx)),
+				true, true);
+	}
+
+	@Test
+	void testIssue1286ConnectionCloseErrorResponse_ServerHTTP11AndH2ClientH2() throws Exception {
+		Http2SslContextSpec serverCtx = Http2SslContextSpec.forServer(ssc.certificate(), ssc.privateKey());
+		Http2SslContextSpec clientCtx =
+				Http2SslContextSpec.forClient()
+				                   .configure(builder -> builder.trustManager(InsecureTrustManagerFactory.INSTANCE));
+		doTestIssue1286(
+				server -> server.protocol(HttpProtocol.H2, HttpProtocol.HTTP11).secure(spec -> spec.sslContext(serverCtx)),
+				client -> client.protocol(HttpProtocol.H2).secure(spec -> spec.sslContext(clientCtx)),
+				true, true);
+	}
+
+	@Test
+	void testIssue1286ConnectionCloseErrorResponse_ServerHTTP11AndH2ClientHTTP11AndH2() throws Exception {
+		Http2SslContextSpec serverCtx = Http2SslContextSpec.forServer(ssc.certificate(), ssc.privateKey());
+		Http2SslContextSpec clientCtx =
+				Http2SslContextSpec.forClient()
+				                   .configure(builder -> builder.trustManager(InsecureTrustManagerFactory.INSTANCE));
+		doTestIssue1286(
+				server -> server.protocol(HttpProtocol.H2, HttpProtocol.HTTP11).secure(spec -> spec.sslContext(serverCtx)),
+				client -> client.protocol(HttpProtocol.H2, HttpProtocol.HTTP11).secure(spec -> spec.sslContext(clientCtx)),
+				true, true);
+	}
+
+	private void doTestIssue1286(
+			Function<HttpServer, HttpServer> serverCustomizer,
+			Function<HttpClient, HttpClient> clientCustomizer,
+			boolean connectionClose, boolean throwException) throws Exception {
+		CountDownLatch latch = new CountDownLatch(1);
+		AtomicReference<List<ByteBuf>> byteBufReplay = new AtomicReference<>(new ArrayList<>());
+		AtomicReference<List<Buffer>> bufReplay = new AtomicReference<>(new ArrayList<>());
+		HttpServer server = serverCustomizer.apply(createServer());
+		disposableServer =
+				server.doOnConnection(conn -> conn.addHandlerLast(new ChannelHandlerAdapter() {
+
+				              @Override
+				              public void channelRead(ChannelHandlerContext ctx, Object msg) {
+				                  if (msg instanceof ByteBufHolder byteBuf) {
+				                      byteBufReplay.get().add(byteBuf.content());
+				                  }
+				                  else if (msg instanceof ByteBuf byteBuf) {
+				                      byteBufReplay.get().add(byteBuf);
+				                  }
+								  else if (msg instanceof Buffer buf) {
+									  bufReplay.get().add(buf);
```

Comment:
There is something wrong with the formatting here ...

Reply:
fixed.